### ISSUE TITLE
Asisted GNSS XML is not decreasing Time-To-First-Fix?

### ISSUE THREAD

**Issue by mmtaksuu** [gnss-sdr/gnss-sdr#691]
Hi, I'm Mehmet.
First of all, I want to thank you for providing us such a huge repository about Software Defined Receiver. 

I've started to learn gnss-sdr via your awesome tutorial [My first position fix](https://gnss-sdr.org/my-first-fix/). I have downloaded **2013_04_04_GNSS_SIGNAL_at_CTTC_SPAIN.tar.gz** file and feed them to the software and then got the position solution.

I've also downloaded GPS Rinex ephemeris file **brdc0940.13n.Z** that belongs to the date of 2013/04/04 from [NASA's Daily GPS](https://cddis.nasa.gov/archive/gnss/data/daily/2013/094/13n/) and then convert them to the XML file format needed by Asisted GNSS using **rinex2assist** executable.

Without giving Asisted GNSS data, the software starts to compute first fixed position solution at **45 s of the receiver time**.
My expactation was to decrease TTFF by providing Asisted GPS data. But it didn't happen. The first fixed position solution time didn't change.

This is [my configuration file](https://pastebin.ubuntu.com/p/6HVHGJpgwh/) that used in my test. Can you tell me what is wrong in my test and what can I do for decreasing TTFF for both in real-time and post-processing mode.

Console output without Asisted GNSS: [console log without AGNSS](https://pastebin.ubuntu.com/p/GMd7Jj3h24/)
Console output with Asisted GNSS: [console log with AGNSS](https://pastebin.ubuntu.com/p/jpmF8rY7xW/)

Thanks in advance for tour help.

**Comment by ValZapod** [comment]
>Asisted GNSS data

That would be usefull to change the direction of the antenna... Raw IQ files or rinex do not have to use any AGNSS data.

>have downloaded 2013_04_04_GNSS_SIGNAL_at_CTTC_SPAIN.tar.gz file and feed them to the software and then got the position solution.

Did you feed it real time?? But it is not real time.

**Comment by ban09** [comment]
> > Asisted GNSS data
> 
> That would be usefull to change the direction of the antenna (either big dish one or a phased array like in smartphones)... Raw IQ files or rinex do not have to use any AGNSS data. I mean they can if you actually capture all, LIKE ALL spectrum there is. With 1000 or so antennas.
> 
> > have downloaded 2013_04_04_GNSS_SIGNAL_at_CTTC_SPAIN.tar.gz file and feed them to the software and then got the position solution.
> 
> Did you feed it real time?? But it is not real time.

I believe the kind of assistance he is referring too is navigation data which as soon as you decode time from one SV should allow you to have a fix without waiting to downlink the navigation message from all satellites and if you have coarse time as well you can do a directed search to the SVs you know are in view reducing further the TTFF.

**Comment by ValZapod** [comment]
Still, while does it take 45 seconds to resolve data that is not REAL time? You can just read it faster than real time from SSD, no? It should be faster on faster PC too. Unless you force it to be real time for some strange reason.

And again, AGNSS allows to capture correct spectrum out of ALL of it, from all 360° and all frequencies. After it is captured, it is not needed.

**Comment by ban09** [comment]
> Still, why does it take 45 seconds to resolve data that is not REAL time? You can just read it faster than real time from SSD, no? It should be faster on faster PC too. Unless you force it to be real time for some strange reason.

It doesn't take 45 seconds, if you are working from a file and your computer is fast it will take less. The receiver time is computed from the number of processed samples times the sampling rate is telling you how far into the file you are, it doesn't have anything to do to the actual time it took to process the file.

Regardless of that, there are instances where you might want to work in pseudo-real time and add a throttle to the system if for instance you are prototyping a system with more components that interact with the receiver.

> And again, AGNSS allows to capture correct spectrum out of ALL of it, from all 360° and all frequencies. After it is captured, it is not needed.

Here I don't know what you mean with this. GNSS-SDR is in the end a research receiver so you might want to work with the same dataset over and over again and try e.g. different settings for the tracking loops or different PVT algorithms. And maybe you want to do this in conjunction with predownloaded assistance data so that you don't need to downlink the navigation message from the satellite (in fact this is what smartphones do most of the time), maybe you don't even want to include the data components.

**Comment by voldrfr1** [comment]
Hi,
I tried hot starting to process OAKBAT dataset with Galileo (https://www.osti.gov/servlets/purl/1836467) and faced the same issue: TTFF not decreasing even after providing the almanac, utc model, ephemeris data, iono model and position+time. It takes around four minutes of data before the first fix, which makes the dataset useless as an event of interest happens at 120 seconds. 

I have tried all different settings and with different datasets hot start seems to work (with TEXBAT it went down from 40s to 20 s). 

Have you resolved your problem? @mmtaksuu
